### PR TITLE
Wrap gRPC errors

### PR DESCRIPTION
**Pull request by geigerj**
Wraps instaces of gRPC AbortionError during RPCs with GaxError.

### CODE REVIEW

File: google/gax/errors.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    Attributes:
+      msg (string): describes the error that occurred.
+      wrapped (Exception, optional): the exception raised by a lower
+        layer of the RPC stack (for example, gRPC) that caused this
+        exception, or None if this exception originated in GAX.
+    """
+    def __init__(self, msg, wrapped=None):
+        super(GaxError, self).__init__(msg)
+        self.msg = msg
```
Comment (tbetbetbe) [inline google/gax/errors.py]: self.msg is unnecessary

File: google/gax/errors.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        self.wrapped = wrapped
+
+    def __str__(self):
+        if not self.wrapped:
+            return self.msg
+        else:
+            return '{} (wraps exception {}: {})'.format(
+                self.msg, type(self.wrapped), self.wrapped)
```
Comment (tbetbetbe) [inline google/gax/errors.py]:
```python
our_msg = super(GaxError, self).__str__()
if not self.cause:
    return our_msg
   
return '{} caused by {}'.format(our_msg, self.cause)
...
```

File: google/gax/errors.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+class GaxError(Exception):
+    """Common base class for exceptions raised by GAX.
+
+    Attributes:
+      msg (string): describes the error that occurred.
+      wrapped (Exception, optional): the exception raised by a lower
+        layer of the RPC stack (for example, gRPC) that caused this
+        exception, or None if this exception originated in GAX.
+    """
+    def __init__(self, msg, wrapped=None):
```
Comment (tbetbetbe) [inline google/gax/errors.py]: s/wrapped/cause/

File: google/gax/errors.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+"""Provides GAX exceptions."""
+
+
+class GaxError(Exception):
+    """Common base class for exceptions raised by GAX.
+
+    Attributes:
+      msg (string): describes the error that occurred.
+      wrapped (Exception, optional): the exception raised by a lower
```
Comment (tbetbetbe) [inline google/gax/errors.py]: s/wrapped/cause/

File: google/gax/errors.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        layer of the RPC stack (for example, gRPC) that caused this
+        exception, or None if this exception originated in GAX.
+    """
+    def __init__(self, msg, wrapped=None):
+        super(GaxError, self).__init__(msg)
+        self.msg = msg
+        self.wrapped = wrapped
```
Comment (tbetbetbe) [inline google/gax/errors.py]: s/wrapped/cause/

File: google/gax/config.py [hunk 6]
```diff
@@ -71,3 +71,7 @@ def exc_to_code(exc):
 
 create_stub = grpc.create_stub  # pylint: disable=invalid-name,
 """The function to use to create stubs."""
+
+
+ABORTION_ERRORS = (face.AbortionError, )
```
Comment (tbetbetbe) [inline google/gax/config.py]: The idea is that it should be possible

```python
import gax.config

gax.config.xxx  = someother_package.yyy  # simple references from another package
```

So this package should just contain assignments from grpc.py, with all the grpc specific import isolated to grpc.py

It's not really part of this CL, but much of the content of this module should be grpc.py:  sorry, I should have picked that up in earlier CLs.

Let's fix that starting with this change.

Move this to grpc.error, and and please use the more generic name API_ERRORS.

File: google/gax/errors.py (hunk 1 above)
Comment (geigerj) [inline google/gax/errors.py]: Changed to using the args attribute inherited from Exception. (I assume that's what you mean?)

File: google/gax/errors.py (hunk 3 above)
Comment (geigerj) [inline google/gax/errors.py]: done

File: google/gax/errors.py (hunk 4 above)
Comment (geigerj) [inline google/gax/errors.py]: done

File: google/gax/errors.py (hunk 5 above)
Comment (geigerj) [inline google/gax/errors.py]: done

File: google/gax/config.py (hunk 6 above)
Comment (geigerj) [inline google/gax/config.py]: Done, and I will follow this change with one to move the rest of config.py into grpc.py

File: google/gax/errors.py (hunk 2 above)
Comment (geigerj) [inline google/gax/errors.py]: Partially done -- do you feel it is not useful to display the type of exception that caused this one? I am concerned the string representation may not be helpful in some cases without the exception type.

Comment (geigerj) [inline google/gax/errors.py]: Done, following the assumption that the string representations of the exceptions we're wrapping are descriptive enough to include their type.

**Comment by tbetbetbe** [comment]
Please bump the version, then I think this is good.

**Comment by geigerj** [comment]
Done

**Pull request merged**